File: BackEnd/Praza-Shop/src/main/java/org/example/prazashop/model/dto/UsuarioDto.java
package org.example.prazashop.model.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.example.prazashop.model.TipoUsuario;

/**
 * The type Usuario dto.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UsuarioDto {
    private Long id;
    private String nome;
    private String email;
    private String contrasinal;
    private String telefono;
    private TipoUsuario tipoUsuario;
}

File: FrontEnd/praza_shop/lib/models/usuario_dto.dart
class UsuarioDto {
  final int? id;
  final String? nome;
  final String? email;
  final String? contrasinal;
  final String? telefono;
  final String? tipoUsuario;

  UsuarioDto({this.id, this.nome, this.email, this.contrasinal, this.telefono, this.tipoUsuario});

  factory UsuarioDto.fromJson(Map<String, dynamic> json) => UsuarioDto(
        id: json['id'] is int ? json['id'] : (json['id'] != null ? int.tryParse('${json['id']}') : null),
        nome: json['nome'],
        email: json['email'],
        contrasinal: json['contrasinal'],
        telefono: json['telefono'],
        tipoUsuario: json['tipoUsuario']?.toString(),
      );

  Map<String, dynamic> toJson() => {
        if (id != null) 'id': id,
        if (nome != null) 'nome': nome,
        if (email != null) 'email': email,
        if (contrasinal != null) 'contrasinal': contrasinal,
        if (telefono != null) 'telefono': telefono,
        if (tipoUsuario != null) 'tipoUsuario': tipoUsuario,
      };
}
